packages/connect-form/src/components/connect-form.tsx

```diff
@@ -44,81 +53,165 @@ const formContentContainerStyles = css({
   padding: spacing[4],
 });
 
+const favoriteButtonStyles = css({
+  position: 'absolute',
+  top: spacing[4],
+  right: spacing[6],
+  hover: {
+    cursor: 'pointer',
+  },
+});
+
+const editFavoriteButtonStyles = css({
+  verticalAlign: 'text-top',
+  marginLeft: spacing[1],
+});
+
+const favoriteButtonLabelStyles = css({
+  position: 'absolute',
+  top: spacing[5],
+  paddingTop: spacing[1],
+  color: uiColors.black,
+  fontWeight: 'bold',
+  fontSize: 12,
+});
+
 function ConnectForm({
   initialConnectionInfo,
   onConnectClicked,
+  // The connect form will not always used in an environment where
+  // the connection info can be saved.
+  showSaveConnection = false,
+  saveConnection,
 }: {
   initialConnectionInfo: ConnectionInfo;
   onConnectClicked: (connectionInfo: ConnectionInfo) => void;
+  showSaveConnection?: boolean;
+  saveConnection: (connectionInfo: ConnectionInfo) => Promise<void>;
```

Comment:
I mentioned this a couple of times already, the usual pattern for callback prop is to be something like `onSaveConnectionClick` instead of an imperative method name, the difference is in what this represent for the code reader / component user: `onSave...` means that you are passing a callback that can do whatever it wants when component activates your callback, `saveConnection` implies what exactly this method does, so if that changes (a contrived example, let's say we change the behavior to immediately connect when save is clicked), you either will be mislead by the method name if you're unaware of the change, or will have to constantly rename it due to flow changes.

Generally speaking I'm totally okay if we don't want to follow the pattern, but I would at least prefer that we keep it consistent in the new code we write, so maybe we can at least rename `onConnectClicked` to `connect` or something

Reply:
Updated to `onSaveConnectionClicked` 👍 makes sense will do in future.